src/test/java/seedu/address/logic/parser/EditCommandParserTest.java

```diff
@@ -42,140 +42,143 @@
-        assertParseFailure(parser, "1" + TAG_EMPTY + TAG_DESC_CORE + TAG_DESC_NON_CORE, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "cs2100" + TAG_DESC_CORE + TAG_DESC_NON_CORE + TAG_EMPTY,
+            Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "cs2100" + TAG_DESC_CORE + TAG_EMPTY + TAG_DESC_NON_CORE,
+            Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "cs2100" + TAG_EMPTY + TAG_DESC_CORE + TAG_DESC_NON_CORE,
+            Tag.MESSAGE_CONSTRAINTS);
 
         // multiple invalid values, but only the first invalid value is captured
-        assertParseFailure(parser, "1" + INVALID_CODE_DESC + VALID_CREDITS_CS2103T,
+        assertParseFailure(parser, "cs2100" + INVALID_CODE_DESC_WITH_PREFIX + VALID_CREDITS_CS2103T,
                 ModuleCode.MESSAGE_CONSTRAINTS);
     }
 
     @Test
     public void parse_allFieldsSpecified_success() {
-        Index targetIndex = INDEX_SECOND_MODULE;
-        String userInput = targetIndex.getOneBased() + CREDITS_DESC_CS3216 + TAG_DESC_NON_CORE
-                + CODE_DESC_CS2103T + TAG_DESC_CORE;
+        ModuleCode targetCode = CODE_SECOND_MODULE;
+        String userInput = targetCode + TAG_DESC_NON_CORE
+                + CODE_DESC_CS2103T_WITH_PREFIX + TAG_DESC_CORE;
 
         EditModuleDescriptor descriptor = new EditModuleDescriptorBuilder().withModuleCode(VALID_CODE_CS2103T)
-                .withModularCredits(VALID_CREDITS_CS3216).withTags(VALID_TAG_CORE, VALID_TAG_NON_CORE).build();
-        EditCommand expectedCommand = new EditCommand(targetIndex, descriptor);
+            .withModuleTitle(VALID_TITLE_CS2103T).withModularCredits(VALID_CREDITS_CS2103T)
+            .withTags(VALID_TAG_CORE, VALID_TAG_NON_CORE).build();
+        EditCommand expectedCommand = new EditCommand(targetCode, descriptor);
 
         assertParseSuccess(parser, userInput, expectedCommand);
     }
 
     @Test
     public void parse_someFieldsSpecified_success() {
-        Index targetIndex = INDEX_FIRST_MODULE;
-        String userInput = targetIndex.getOneBased() + CREDITS_DESC_CS3216 + CODE_DESC_CS2103T;
+        ModuleCode targetCode = CODE_FIRST_MODULE;
+        String userInput = targetCode + CODE_DESC_CS2103T_WITH_PREFIX;
 
-        EditModuleDescriptor descriptor = new EditModuleDescriptorBuilder().withModularCredits(VALID_CREDITS_CS3216)
-                .withModuleCode(VALID_CODE_CS2103T).build();
-        EditCommand expectedCommand = new EditCommand(targetIndex, descriptor);
+        EditModuleDescriptor descriptor = new EditModuleDescriptorBuilder()
+            .withModuleCode(VALID_CODE_CS2103T).withModuleTitle(VALID_TITLE_CS2103T)
+            .withModularCredits(VALID_CREDITS_CS2103T).build();
+        EditCommand expectedCommand = new EditCommand(targetCode, descriptor);
 
         assertParseSuccess(parser, userInput, expectedCommand);
     }
 
     @Test
     public void parse_oneFieldSpecified_success() {
```

Comment:
Smol nit: Since this test and the one before it both only specify one field, should we name them "onlyModuleCodeSpecified" and "onlyTagsSpecified" or something like that instead?

Reply:
done!